### PR TITLE
[I-199][BR1028][IMP]Improved Project Creation Process

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
Issue #199 

- Project Created BR with RL,without DL

### CODE REVIEW

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS functional tests failed:
I created one BR with 1 DL (containing 2 RL) and 1 extra RL.
Created the project: 2 tasks are created
Expected: 3 tasks are created

**Comment by YogeshMahera-SerpentCS** [comment]
Yes, the Improved code, could you please review.
@elicoidal

**Review by elicoidal — APPROVED** [review]
@YogeshMahera-SerpentCS Tests LGTM
@victormartinelicocorp

**Comment by elicoidal** [comment]
@dreispt @pedrobaeza Can you help?

File: business_requirement_deliverable_project/models/project.py [hunk 1]
```diff
@@ -41,7 +41,16 @@ def generate_project_wizard(self):
...
-
+            for resource_line in br.resource_lines:
+                if resource_line.resource_type != 'task':
+                    continue
+                generated = self.env['project.task'].search(
+                    [('br_resource_id', '=', resource_line.id)],
+                    limit=1)
+                if generated:
+                    continue
+                lines.append(resource_line.id)
```
Comment (pedrobaeza) [inline business_requirement_deliverable_project/models/project.py]: Use a recordset instead:
```
lines = env['...']
...
lines |= resource_line
```

File: business_requirement_deliverable_project/wizard/generate_projects.py [hunk 2]
```diff
@@ -121,12 +121,8 @@ def generate_br_projects(self, parent_project, br, project_ids, task_ids):
             else:
                 br_project = br.linked_project
                 project_ids.append(br_project.id)
-            if not self.for_deliverable:
-                lines = [
-                    line.resource_ids for line in br.deliverable_lines
-                    if line.resource_ids
-                ]
-                self.create_project_task(lines, br_project.id, task_ids)
+            lines = [br.resource_lines]
```
Comment (pedrobaeza) [inline business_requirement_deliverable_project/wizard/generate_projects.py]: Why encapsulating this in another list?

File: business_requirement_deliverable_project/models/project.py [hunk 3]
```diff
@@ -41,7 +41,16 @@ def generate_project_wizard(self):
                     if generated:
                         continue
                     lines.append(line.id)
-
+            for resource_line in br.resource_lines:
```
Comment (pedrobaeza) [inline business_requirement_deliverable_project/models/project.py]:
```
for resource_line in br.resource_lines.filtered(
    lamba x: x.resource_type == 'task'
):
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS @victormartinelicocorp

**Comment by martintorresvm** [comment]
@YogeshMahera-SerpentCS Any problem to fix the code with @pedrobaeza review?

**Comment by martintorresvm** [comment]
@YogeshMahera-SerpentCS Please increase coveralls

**Comment by elicoidal** [comment]
@pedrobaeza can we move forward?

**Comment by elicoidal** [comment]
@victormartinelicocorp I think coverall is decent enough

**Review by pedrobaeza — APPROVED** [review]

**Comment by elicoidal** [comment]
Thanks @pedrobaeza !

**Pull request merged**